src/openLibrary/producer/parser.js

```diff
@@ -0,0 +1,402 @@
+	work.source = 'OPENLIBRARY';
+
+	// Set up aliases
+	// The first alias (in the logical flow) is made to be default alias
+	let defaultAlias = true;
+	if (!isNotDefined(json.title)) {
+		const name = json.title;
+		const lang = detectLanguage(name);
+
+		work.alias.push({
+			default: defaultAlias,
+			languageId: lang,
+			name,
+			primary: true,
+			sortName: sortName(name)
+		});
+		defaultAlias = false;
+	}
+	if (!isNotDefined(json.subtitle)) {
+		const name = json.subtitle;
+		const lang = detectLanguage(name);
+
+		work.alias.push({
+			default: defaultAlias,
+			languageId: lang,
+			name,
+			sortName: sortName(name)
+		});
+		defaultAlias = false;
+	}
+
+
+	// Last Edited at source
+	if (!isNotDefined(_.get(json, 'last_modified.value'))) {
+		work.lastEdited = json.last_modified.value;
+	}
+
+	if (!isNotDefined(json.key)) {
+		const openLibraryWorkId = json.key.split('/')[2];
+		work.identifiers.push({
+			typeId: identifiers.openLibraryWorkId,
+			value: openLibraryWorkId
+		});
+		work.originId = openLibraryWorkId;
+	}
+
+	if (!isNotDefined(json.authors) && (json.authors instanceof Array)) {
+		json.authors.forEach((authorObj) => {
+			if (!isNotDefined(authorObj && _.get(authorObj, 'author.key'))) {
+				work.metadata.relationships.push({
+					type: 'authoredBy',
+					value: authorObj.author.key.split('/')[2]
+				});
+			}
+		});
+	}
+
+	if (!isNotDefined(json.description)) {
+		work.disambiguation = json.description.value;
+	}
+
+	if (!isNotDefined(json.links) && (json.links instanceof Array)) {
+		json.links.forEach(link => {
+			if (!isNotDefined(link.title && link.url)) {
+				work.metadata.links.push({
+					title: link.title,
+					url: link.url
+				});
+			}
+		});
+	}
+
+	const metadataFields = [
+		'subjects',
+		'subject_places',
+		'subject_people',
+		'description',
+		'subject_times',
+		'cover_edition',
+		'works',
+		'lc_classifications',
+		'first_publish_date',
+		'dewey_number',
+		'first_sentence',
+		'excerpts',
+		'number_of_editions',
+		'remote_ids'
+	];
+
+	metadataFields.forEach(field => {
+		if (!isNotDefined(json[field])) {
+			work.metadata[field] = json[field];
+		}
+	});
+
+	return work;
+}
+
+function processAuthor(json) {
+	if (isNotDefined(json)) {
+		return null;
+	}
+
+	const creator = {};
+	creator.alias = [];
+	creator.identifiers = [];
+
+	creator.metadata = {};
+	creator.metadata.relationships = [];
+	creator.metadata.links = [];
+	creator.metadata.originId = [];
+	creator.metadata.identifiers = [];
+
+	creator.entityType = entityTypes.CREATOR;
+	creator.source = 'OPENLIBRARY';
+
+	// Set up aliases
+	// The first alias (in the logical flow) is made to be default alias
+	let defaultAlias = true;
+	if (!isNotDefined(json.name)) {
+		const {name} = json;
+		const lang = detectLanguage(name);
+
+		creator.alias.push({
+			default: defaultAlias,
+			languageId: lang,
+			name,
+			sortName: sortName(name)
+		});
+		defaultAlias = false;
+	}
+	if (!isNotDefined(json.personal_name)) {
```

Comment:
I'm not sure about this - what is the difference between name, personal name and fuller name?

Reply:
The data dumps samples appeared to have just similar names. I myself am doubtful about what these actually signify. There's no documentation which we can reference.